src/core/primitives.rs

```diff
@@ -41,6 +41,7 @@ fn compute_user_key_kmac(msk: &MasterSecretKey, usk: &UserSecretKey) -> Option<K
     user_key_bytes.extend_from_slice(&usk.a.to_bytes());
     user_key_bytes.extend_from_slice(&usk.b.to_bytes());
 
+    // KMAC is deterministic because subkeys is a Vec preserving keys order
```

Comment:
How does this comment help understanding the code here? Wouldn't it make more sense somewhere else (e.g. to justify the use of a `Vec` for the list of subkeys)? Is it necessary?